dashboard/app/controllers/api/v1/pd/international_opt_ins_controller.rb

```diff
@@ -15,5 +17,7 @@ def on_successful_create
       source: EmailPreference::FORM_PD_INTERNATIONAL_OPT_IN,
       form_kind: "0"
     )
+
+    Services::Teacher.verify(current_user)
```

Comment:
I thought `permission` was an attribute of the user record, but it's not. Considering this, `User#verify_teacher!` is all we need as it saves the permission we set and raises an error if it is not a teacher account.
https://github.com/code-dot-org/code-dot-org/blob/419f8e642df0b9a170a318867f866301ba489915/dashboard/app/models/concerns/user_permission_grantee.rb#L28-L32
```suggestion
    current_user.verify_teacher! if current_user.teacher?
```

Reply:
To be sure I'm reading correctly, the `find_or_create_by` won't create a new permission record if the permission already exists, and no email will be sent since that's handle in the after_create hook?
https://github.com/code-dot-org/code-dot-org/blob/054fae1799ef1e74c8bcb14db8ae11bf116e00d9/dashboard/app/models/user_permission.rb#L60

Reply:
It's correct